tests/datasource/fluent/data_asset/test_sql_asset.py

```diff
@@ -168,3 +232,83 @@ def test_add_batch_definition_fluent_sql__add_batch_definition_daily(
     # assert
     assert batch_definition == expected_batch_definition
     datasource.add_batch_definition.assert_called_once_with(expected_batch_definition)
+
+
+class ExpectNoError:
+    def __enter__(self): ...
+
+    def __exit__(self, exc_type, exc_val, exc_tb): ...
+
+
+def add_table_asset(datasource: SQLDatasource):
+    return datasource.add_table_asset(
+        name="my_table_asset",
+        table_name="my_table",
+        schema_name="my_schema",
+    )
+
+
+def add_query_asset(datasource: SQLDatasource):
+    return datasource.add_query_asset(name="my_query_asset", query="select * from my_table")
+
+
+@pytest.mark.unit
+@pytest.mark.parametrize(
+    "queried_row, expect_error_or_no",
+    [
+        pytest.param(
+            [datetime(2024, 10, 28, 0, 0, 0, tzinfo=timezone.utc)],
+            lambda: ExpectNoError(),
```

Comment:
Non-blocking, but I'd like this to be considered: I find this test setup with the no-op context manager kinda confusing. What do you think about making 2 tests: one for all the cases where an error is raised, and one for the happy path?